icu4c/source/common/locid.cpp

```diff
@@ -1331,6 +1331,9 @@ class UnicodeKeywordEnumeration : public KeywordEnumeration {
     }
 };
 
+// Out-of-line virtual destructor to serve as the "key function".
+UnicodeKeywordEnumeration::~UnicodeKeywordEnumeration() {}
```

Comment:
Is there any reason to change from `= default;` to `{}` here?

Reply:
Yes, so that the tool can see that the class destructor serves as its "key function", as mentioned in the PR description. See also "Virtual Destructors" on http://userguide.icu-project.org/dev/codingguidelines

Reply:
Apparently "= default" works in an out-of-line definition as well, so that's fine. We just can't implement the virtual destructor inside the class declaration, no matter default vs. real.